core/runtime/src/main/java/io/quarkiverse/jberet/runtime/JBeretConfig.java

```diff
@@ -58,7 +58,7 @@ interface Repository {
          *
          */
         @WithDefault("IN_MEMORY")
-        Type type();
+        String type();
```

Comment:
I'm not against removing the enum. I know it can be problematic when we have dynamic registrations, but without it, there is no way to validate that the user is setting up a valid repository.

If we remove the enum, we should validate that the repository type value matches any of the available repositories.

Reply:
We can keep the enum, if you want.
But keep in mind that this implies modifying the core runtime everytime we want to add another job repository type.
With my approach repository types are dynamically discovered and can be loaded from external libraries.

Please let me know how you want me to proceed.

Maybe we could add a check in the JBeretRecorder and validate the repository type, as you suggest.
When using full CDI I can retrieve an Instance call for the isResolvable method.
Using Quarkus BeanManager how can this be done ?
There is no isResolvable in this implementation ...

Help ?

Reply:
Ideally, you want to do this during build time.

We could add a new build item that contains the bean the Repository type to register with Arc. If the build item is unavailable, it would mean that no registration took place, so no valid repositories were found, and you have an issue with the configuration.

Reply:
Done !